numpy/ma/core.pyi

```diff
@@ -408,23 +408,23 @@ class MaskedArray(ndarray[_ShapeT_co, _DTypeT_co]):
     def shape(self) -> _ShapeT_co: ...
     @shape.setter
     def shape(self: MaskedArray[_ShapeT, Any], shape: _ShapeT, /) -> None: ...
-    def __setmask__(self, mask, copy=...): ...
+    def __setmask__(self, mask: _ArrayLikeBool_co, copy: bool = False) -> None: ...
     @property
-    def mask(self): ...
+    def mask(self) -> NDArray[bool_] | MaskType: ...
     @mask.setter
-    def mask(self, value): ...
+    def mask(self, value: _ArrayLikeBool_co) -> None: ...
     @property
     def recordmask(self): ...
     @recordmask.setter
     def recordmask(self, mask): ...
-    def harden_mask(self): ...
-    def soften_mask(self): ...
+    def harden_mask(self) -> Self: ...
```

Comment:
Usually I prefer to group `@properties` together, but I can see how in this case it makes sense to keep it like this. So dealer's choice 🤷🏻